apps/wallet/src/ui/app/helpers/errorMessages.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { LockedDeviceError, StatusCodes } from '@ledgerhq/errors';
+
+import {
+    isLedgerTransportStatusError,
+    LedgerConnectionFailedError,
+    LedgerDeviceNotFoundError,
+    LedgerNoTransportMechanismError,
+} from '../components/ledger/ledgerExceptions';
+
+/**
+ * Helper method for producing user-friendly error messages from Signer operations
+ * from SignerWithProvider instances (e.g., signTransaction, getAddress, and so forth)
+ */
+export function getSignerOperationErrorMessage(error: unknown) {
+    return (
+        getLedgerConnectionErrorMessage(error) ||
+        getSuiApplicationErrorMessage(error) ||
+        (error as Error).message ||
+        'Something went wrong.'
+    );
+}
+
+/**
+ * Helper method for producing user-friendly error messages from Ledger connection errors
+ */
+export function getLedgerConnectionErrorMessage(error: unknown) {
```

Comment:
we're mapping errors -> UI messages since internal error messages are usually more descriptive and we don't want to couple those to the UI